burnman/mademember.py

```diff
@@ -0,0 +1,288 @@
+        Aliased with self.K_S
+        """
+        return 1. / self.adiabatic_bulk_modulus
+
+    @material_property
+    def p_wave_velocity(self):
+        """
+        Returns P wave speed of the solid solution [m/s]
+        Aliased with self.v_p
+        """
+        return np.sqrt((self.adiabatic_bulk_modulus + 4. / 3. *
+                        self.shear_modulus) / self.density)
+
+    @material_property
+    def bulk_sound_velocity(self):
+        """
+        Returns bulk sound speed of the solid solution [m/s]
+        Aliased with self.v_phi
+        """
+        return np.sqrt(self.adiabatic_bulk_modulus / self.density)
+
+    @material_property
+    def shear_wave_velocity(self):
+        """
+        Returns shear wave speed of the solid solution [m/s]
+        Aliased with self.v_s
+        """
+        return np.sqrt(self.shear_modulus / self.density)
+
+    @material_property
+    def grueneisen_parameter(self):
+        """
+        Returns grueneisen parameter of the solid solution [unitless]
+        Aliased with self.gr
+        """
+        if self.temperature < 1.e-12:
+            return 0.
+        else:
+            return self.thermal_expansivity * self.isothermal_bulk_modulus \
+                * self.molar_volume / self.heat_capacity_v
+
+    @material_property
+    def heat_capacity_v(self):
+        """
+        Returns heat capacity at constant volume of the solid solution [J/K/mol]
+        Aliased with self.C_v
+        """
+        return self.heat_capacity_p - self.molar_volume * self.temperature \
+            * self.thermal_expansivity * self.thermal_expansivity \
+            * self.isothermal_bulk_modulus
+
+
+def make_endmember(mineral_list, molar_amounts, free_energy_adjustment=[]):
+    """
+    This function makes a new endmember from a linear combination of minerals
+    and a Gibbs free energy adjustment which is linear in
+    pressure and temperature (i.e. a constant energy, entropy and
+    volume adjustment).
+    
+    For example, a crude approximation to a bridgmanite model might be
+    bdg = make_endmember([per, stv], [1.0, 1.0], [-15.e3, 0., 0.])
+
+
+    Parameters
+    ----------
+    mineral_list : list of instances of class Mineral
+        Minerals from which the linear combination is to be constructed
+
+    molar_amounts : list of floats
+        Stoichiometric coefficients for the linear combination
+        These can be negative.
+
+    free_energy_adjustment : list of floats (optional)
+        This list of three floats should contain constant 
+        internal energy, entropy and volume modifiers. 
+        These should be given in SI units.
+
+    Returns
+    -------
+    m : Instance of MadeMember class
+        The desired mineral 
+    """
+    assert(len(free_energy_adjustment) == 3)
+
+    m = MadeMember(mineral_list, molar_amounts)
+
+    if free_energy_adjustment != []:
```

Comment:
Seems like this check is already asserted above. 
Why is this separate function needed? Most things I 'make' by initializing them, why is this constructed differently? If the only difference is that I want to be able to set them without free energy adjustments, this could just be an optional argument.